# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JSAtom.cpp
# Commit: 73928903648a
# Full Hash: 73928903648aa79ca0b4fa0bb27ae844c34577e5
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-08-24 21:34:05
# Regressor Bug: 1786506
# File Overlap Count: 1
# Description:
#   Bug 1786506 - Part 2: Give shared permanent things their own zone r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155378
# ==============================================================================

diff -r 747154213937 -r 73928903648a js/src/vm/JSAtom.cpp
--- a/js/src/vm/JSAtom.cpp	Wed Aug 24 09:43:06 2022 +0000
+++ b/js/src/vm/JSAtom.cpp	Wed Aug 24 09:43:06 2022 +0000
@@ -214,10 +214,6 @@
     return bool(atoms_);
   }
 
-#ifdef DEBUG
-  gc.assertNoPermanentSharedThings();
-#endif
-
   // NOTE: There's no GC, but `gc.freezePermanentSharedThings` below contains
   //       a function call that's marked as "Can GC".
   Rooted<UniquePtr<AtomSet>> atomSet(cx,
@@ -312,7 +308,9 @@
     wellKnownSymbols = wks;
   }
 
-  gc.freezePermanentSharedThings();
+  if (!gc.freezeSharedAtomsZone()) {
+    return false;
+  }
 
   // The permanent atoms table has now been populated.
   permanentAtoms_ =
